menas/ui/components/mappingTable/mappingTableMain.controller.js

```diff
@@ -13,415 +13,328 @@
-  },
+    onSchemaSelect: function (oEv) {
+      this._editFragment.onSchemaSelect(oEv);
+    },
 
-  onAfterRendering : function() {
-    // get schemas after rendering. This will be used for add/edit
-    // functionality
-    let schemas = this._model.getProperty("/schemas")
-    if (!schemas || schemas.length === 0) {
-      SchemaService.getSchemaList(false, true);
-    }
-  },
-
-  onEditPress : function() {
-    let current = this._model.getProperty("/currentMappingTable");
-
-    current.isEdit = true;
-    current.title = "Edit";
-
-    this._model.setProperty("/newMappingTable", jQuery.extend(true, {}, current));
-
-    SchemaService.getAllSchemaVersions(current.schemaName, sap.ui.getCore().byId("newMappingTableSchemaVersionSelect"))
-
-    this._addDialog.open();
-  },
-
-  onAddPress : function() {
-    let oFirstSchema = this._model.getProperty("/schemas")[0];
-
-    this._model.setProperty("/newMappingTable", {
-      name: "",
-      description: "",
-      schemaName: oFirstSchema._id,
-      schemaVersion: oFirstSchema.latestVersion,
-      hdfsPath: "/",
-      isEdit : false,
-      title : "Add"
-    });
-
-    this._addDialog.open();
-  },
-
-  schemaSelect : function(oEv) {
-    let sSchemaId = oEv.getParameter("selectedItem").getKey()
-    SchemaService.getAllSchemaVersions(sSchemaId, sap.ui.getCore().byId("newMappingTableSchemaVersionSelect"),
-      this._model, "/newMappingTable/schemaVersion")
-  },
-
-  mappingTableSelected : function(oEv) {
-    let selected = oEv.getParameter("listItem").data("id")
-    this._router.navTo("mappingTables", {
-      id : selected
-    });
-  },
-
-  mappingTableNameChange : function() {
-    let sName = this._model.getProperty("/newMappingTable/name");
-    if (GenericService.isValidEntityName(sName)) {
-      MappingTableService.hasUniqueName(sName)
-    } else {
-      this._model.setProperty("/newMappingTable/nameUnique", true)
-    }
-  },
-
-  onRemovePress : function(oEv) {
-    let current = this._model.getProperty("/currentMappingTable")
-
-    sap.m.MessageBox.show("This action will remove ALL versions of the mapping table definition. \nAre you sure?.", {
-      icon : sap.m.MessageBox.Icon.WARNING,
-      title : "Are you sure?",
-      actions : [ sap.m.MessageBox.Action.YES, sap.m.MessageBox.Action.NO ],
-      onClose : function(oAction) {
-        if (oAction == "YES") {
-          MappingTableService.disableMappingTable(current.name)
+    onAfterRendering: function () {
+      // get schemas after rendering. This will be used for add/edit functionality
+      let schemas = this._model.getProperty("/schemas");
+      if (!schemas || schemas.length === 0) {
+        SchemaService.getSchemaList(false, true);
+      }
+    },
+
+    onRemovePress: function (oEv) {
+      let current = this._model.getProperty("/currentMappingTable")
+
+      MessageBox.show("This action will remove ALL versions of the mapping table definition. \nAre you sure?.", {
```

Comment:
Same as before `dot` after `question mark`
Second thing is, why do we have all caps `ALL` here and not up there.